app/Http/Controllers/BrevoController.php

```diff
@@ -61,12 +64,157 @@ public function __construct()
+     *             'X-Gm-Message-State' => 'AOJu0YzKhR1HY1oUXoq++LLpl6UOz1S60NfPxuPXBLcP+6aACYle8rqQ fYHe2rQYTpg4KWiOswu858STOW8qmiewXD6gH/LbmEFs7sknRyDPNr/+L0cv828A3o+SOvXu3uP SY6H1aNSwIpqTRhJ+nNjTuSUpuSoABd9fYXFwPuivV0DtBhoVmpE=',
+     *             'X-Google-Smtp-Source' => 'AGHT+IHdA9ZhW0dQxgOYx2OXBGmu4pzSR/zwJ0vcPNXFSqttKCPS2oTw1a9b2mMdhyUeoRAwP5TmhHlAtqUUrOPwkgg=',
+     *             'X-Received' => 'by 2002:a50:d74c:0:b0:567:3c07:8bbc with SMTP id i12-20020a50d74c000000b005673c078bbcmr2126401edj.21.1711214316135; Sat, 23 Mar 2024 10:18:36 -0700 (PDT)',
+     *             'MIME-Version' => '1.0',
+     *             'From' => 'Max Mustermann <max@mustermann.de>',
+     *             'Date' => 'Sat, 23 Mar 2024 18:18:20 +0100',
+     *             'Message-ID' => '<CADfEuNvumhUdqAUa0j6MxzVp0ooMYqdb_KZ7nZqHNAfdDqwWEQ@mail.gmail.com>',
+     *             'Subject' => 'TEST',
+     *             'To' => 'test@test.de',
+     *             'Content-Type' => 'multipart/mixed',
+     *         ),
+     *         'SpamScore' => 2.8,
+     *         'ExtractedMarkdownMessage' => 'TEST',
+     *         'ExtractedMarkdownSignature' => NULL,
+     *         'RawHtmlBody' => '<div dir="ltr">TEST</div>',
+     *         'RawTextBody' => 'TEST',
+     *         'EMLDownloadToken' => 'eyJmb2xkZXIiOiIyMDI0MDMyMzE3MTgzNi45OS43OTgwMDM4MDQiLCJmaWxlbmFtZSI6InNtdHAuZW1sIn0',
+     *         ),
+     *     ),
+     *   )
+     */
+    public function inboundWebhook(Request $request)
+    {
+        $input = $request->all();
+
+        // validation for client mail credentials by recipient
+        if ($request->has('company')) {
+            if (!($request->has('token')))
+                return response()->json(['message' => 'Unauthorized'], 403);
+
+            $company = MultiDB::findAndSetDbByCompanyId($request->has('company'));
+            $company_brevo_secret = $company->settings?->email_sending_method === 'client_brevo' && $company->settings?->brevo_secret ? $company->settings?->brevo_secret : null;
+            if (!$company || $request->get('token') !== $company_brevo_secret)
+                return response()->json(['message' => 'Unauthorized'], 403);
+
+        } else if (!($request->has('token') && $request->get('token') == config('services.brevo.secret')))
+            return response()->json(['message' => 'Unauthorized'], 403);
+
```

Comment:
@turbo124 in brevo webhook we will use validation by providing the secret used by the api within the url. when using the endpoint with client credentials, we have to set the db first. (as implemented) 
should we use companyId or companyKey in the request?

should we used a hashed version of the api-key for security reasons?

Reply:
@paulwer definitely company_key